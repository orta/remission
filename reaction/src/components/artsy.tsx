import * as PropTypes from "prop-types"
import * as React from "react"

/**
 * The Artsy specific props injected by the higher-order component produced by `ContextConsumer`.
 *
 * @see {@link ContextProvider}
 * @see {@link ContextConsumer}
 *
 * @todo Once this PR for `rest` https://github.com/Microsoft/TypeScript/pull/13470 lands weâ€™ll be able to not make this
 *       optional and simply remove it from the props that a component wrapped with the `ContextConsumer` HOC accepts.
 */
export interface ContextProps {
  /**
   * The currently signed-in user.
   */
  currentUser?: User
}

interface PrivateContextProps extends ContextProps {
  /**
   * This prop solely exists to verify that a consumer is nested inside a provider.
   */
  _isNestedInProvider: boolean
}

const ContextTypes: React.ValidationMap<PrivateContextProps> = {
  _isNestedInProvider: PropTypes.bool,
  currentUser: PropTypes.object,
}

/**
 * This component provides Artsy specific context props to components down the tree. Components that wish to consume
 * these props should be wrapped with the `ContextConsumer` higher-order component.
 *
 * @see {@link ContextConsumer}
 */
export class ContextProvider extends React.Component<ContextProps, null>
  implements React.ChildContextProvider<PrivateContextProps> {
  static childContextTypes = ContextTypes

  constructor(props) {
    if (React.Children.count(props.children) > 1) {
      throw new Error("A ContextProvider expects a single child.")
    }
    super(props)
  }

  getChildContext() {
    return {
      _isNestedInProvider: true,
      currentUser: this.props.currentUser,
    }
  }

  render() {
    return this.props.children as any
  }
}

/**
 * Wraps a component in a higher-order component that injects the Artsy specific context props into the given component.
 * Wrapped components are expected to be rendered in a tree that contains a `ContextProvider` component.
 *
 * The injected props are described in `ContextProps`.
 *
 * @see {@link ContextProvider}
 * @see {@link ContextProps}
 *
 * @param Component The component that needs Artsy specific context props.
 */
export function ContextConsumer<P>(
  Component: React.ComponentClass<P> | React.StatelessComponent<P>
): React.ComponentClass<P> {
  const name = Component.displayName || Component.name
  return class extends React.Component<P, void> {
    static contextTypes = ContextTypes
    static displayName = `Artsy(${name})`

    constructor(props, context: PrivateContextProps) {
      if (!context._isNestedInProvider) {
        const start = name || "A component"
        throw new Error(`${start}, which needs Artsy props, was not wrapped inside a ContextProvider component.`)
      }
      super(props, context)
    }

    render() {
      const currentUser = this.context.currentUser
      const props = Object.assign({ currentUser }, this.props)
      return <Component {...props} />
    }
  }
}
