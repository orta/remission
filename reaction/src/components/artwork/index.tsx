import * as React from "react"
import styled, { css } from "styled-components"

import theme from "../../assets/theme"
import ArtworkMetadata, { ArtworkMetadataProps } from "./metadata"
import createContainer, { RelayProps } from "./relay"

const Container = styled.div`
  width: 100%;
`

const ImageContainer = styled.div`
  width: 100%;
  position: relative;
  cursor: pointer;

  &:before {
    content: '';
    display: block;
    padding-bottom: 100%;
  }

  & .overlay-container {
    transition: opacity 0.25s;

    &.hovered {
      opacity: 0;
      visibility: hidden;
    }

    &.selected {
      opacity: 1;
      visibility: visible;
    }
  }

  &:hover {
    & .overlay-container.hovered {
      ${css`
        @media (min-width: ${theme.flexboxgrid.breakpoints.sm + 1}px) {
          opacity: 1;
          visibility: visible;
        }
      `}
    }
  }
`

const Image = styled.img`
  max-width: 100%;
  max-height: 100%;
  margin: auto;
  position: absolute;
  left: 0;
  top: 0;
  bottom: 0;
  right: 0;
`

export interface OverlayProps {
  selected: boolean
}

export interface ArtworkProps extends RelayProps {
  size?: number
  extended?: boolean
  Overlay?: React.SFC<OverlayProps>
  onSelect?: (selected: boolean) => void
  showOverlayOnHover?: boolean
}

export interface ArtworkState {
  isSelected: boolean
}

export class Artwork extends React.Component<ArtworkProps, ArtworkState> {
  static defaultProps = {
    size: 250,
    extended: true,
    overlay: null,
    showOverlayOnHover: false,
  }

  constructor(props) {
    super(props)
    this.state = {
      isSelected: false,
    }
  }

  onSelected(e) {
    if (!this.props.Overlay) {
      return
    }

    this.setState({
      isSelected: !this.state.isSelected,
    })

    if (this.props.onSelect) {
      this.props.onSelect(!this.state.isSelected)
    }
  }

  render() {
    const { size, artwork, Overlay, showOverlayOnHover } = this.props
    let overlayClasses = "overlay-container"

    overlayClasses += showOverlayOnHover ? " hovered" : ""
    overlayClasses += this.state.isSelected ? " selected" : ""

    return (
      <Container onClick={this.onSelected.bind(this)} size={size}>
        <ImageContainer size={size}>
          <Image src={artwork.image.url} size={size} />
          <div className={overlayClasses}>
            {Overlay && <Overlay selected={this.state.isSelected} />}
          </div>
        </ImageContainer>
        <ArtworkMetadata artwork={artwork} />
      </Container>
    )
  }
}

export default createContainer<ArtworkProps, ArtworkMetadataProps>(Artwork, ArtworkMetadata)
