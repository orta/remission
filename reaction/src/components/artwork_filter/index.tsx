import * as React from "react"
import * as Relay from "react-relay"
import styled from "styled-components"

import Artworks from "../artwork_grid"
import BorderedPulldown from "../bordered_pulldown"
import Spinner from "../spinner"

import Dropdown from "./dropdown"
import ForSaleCheckbox from "./for_sale_checkbox"
import Headline from "./headline"
import TotalCount from "./total_count"

const PageSize = 10

interface Props extends RelayProps, React.HTMLProps<ArtworkFilter> {
  filter_artworks: any
  relay: any
  for_sale?: boolean
  dimension_range?: string
  price_range?: string
  medium?: string
}

interface State {
  for_sale: boolean
  dimension_range: string
  price_range: string
  medium: string
  loading: boolean
}

class ArtworkFilter extends React.Component<Props, State> {
  constructor(props) {
    super(props)
    this.state = {
      for_sale: props.for_sale || false,
      dimension_range: props.dimension_range || "*",
      price_range: props.price_range || "*",
      medium: props.medium || "*",
      loading: false,
    }
  }

  handleLoadMore() {
    if (!this.state.loading) {
      this.setState({ loading: true }, () => {
        this.props.relay.setVariables(
          {
            size: this.props.relay.variables.size + PageSize,
          },
          readyState => {
            if (readyState.done) {
              this.setState({ loading: false })
            }
          }
        )
      })
    }
  }

  setForSale() {
    const isForSale = !this.state.for_sale
    const forSaleVar = isForSale ? true : null

    this.setState({
      for_sale: isForSale,
    })
    this.props.relay.setVariables({
      for_sale: forSaleVar,
      size: PageSize,
    })
  }

  onSelect(count, slice) {
    this.setState({
      [slice.toLowerCase()]: count.id,
    })
    this.props.relay.setVariables({
      [slice.toLowerCase()]: count.id,
      size: PageSize,
    })
  }

  onChangeSort(option) {
    this.props.relay.setVariables({
      sort: option.val,
      size: PageSize,
    })
  }

  render() {
    const filterArtworks = this.props.filter_artworks.filter_artworks
    const dropdowns = filterArtworks.aggregations.map(aggregation =>
      <Dropdown
        aggregation={aggregation}
        key={aggregation.slice}
        onSelect={(count, slice) => this.onSelect(count, slice)}
      />
    )
    const pulldownOptions = [
      { val: "-partner_updated_at", name: "Recently Updated" },
      { val: "-year", name: "Artwork Year (desc.)" },
      { val: "year", name: "Artwork Year (asc.)" },
    ]

    return (
      <div>
        <FilterBar>
          <ForSaleCheckbox checked={this.state.for_sale} onClick={() => this.setForSale()} />
          {dropdowns}
        </FilterBar>
        <SubFilterBar>
          <div>
            <Headline
              medium={this.state.medium}
              price_range={this.state.price_range}
              dimension_range={this.state.dimension_range}
              for_sale={this.state.for_sale}
              facet={filterArtworks.facet}
              aggregations={filterArtworks.aggregations}
            />
            <TotalCount filter_artworks={filterArtworks} />
          </div>
          <BorderedPulldown
            defaultValue="Recently Updated"
            options={pulldownOptions}
            onChange={option => this.onChangeSort(option)}
          />
        </SubFilterBar>
        <Artworks artworks={filterArtworks.artworks} onLoadMore={() => this.handleLoadMore()} columnCount={4} />
        <SpinnerContainer>
          {this.state.loading ? <Spinner /> : ""}
        </SpinnerContainer>
      </div>
    )
  }
}

const FilterBar = styled.div`
  text-align: center;
`

const SubFilterBar = styled.div`
  display: flex;
  justify-content: space-between;
  padding: 40px 0 20px;
  align-items: center;
`

const SpinnerContainer = styled.div`
  width: 100%;
  height: 200px;
  position: relative;
`

export default Relay.createContainer(ArtworkFilter, {
  initialVariables: {
    sort: "-partner_updated_at",
    size: PageSize,
    for_sale: null,
    medium: "*",
    aggregations: ["MEDIUM", "TOTAL", "PRICE_RANGE", "DIMENSION_RANGE"],
    price_range: "*",
    dimension_range: "*",
    gene_id: null,
    tag_id: null,
    artist_id: null,
  },
  fragments: {
    filter_artworks: () => Relay.QL`
      fragment on Viewer {
        filter_artworks(
          aggregations: $aggregations, 
          size: $size,
          for_sale: $for_sale,
          medium: $medium,
          price_range: $price_range,
          dimension_range: $dimension_range,
          sort: $sort,
          gene_id: $gene_id,
          tag_id: $tag_id,
          artist_id: $artist_id,
        ) {
          ${TotalCount.getFragment("filter_artworks")}
          aggregations {
            slice 
            counts {
              id
              name
            }
            ${Dropdown.getFragment("aggregation")}
          }
          artworks: artworks_connection(first: $size) {
            ${Artworks.getFragment("artworks")}
          }
          facet {
            ${Headline.getFragment("facet")}
          }
        }
      }
    `,
  },
})

interface RelayProps {
  filter_artworks: {
    filter_artworks: {
      artworks: Array<any | null> | null
      counts: {
        total: number | null
      } | null
      aggregations: Array<any | null> | null
    } | null
  } | null
}
