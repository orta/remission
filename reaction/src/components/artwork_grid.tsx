import * as React from "react"
import * as ReactDOM from "react-dom"
import * as Relay from "react-relay"

import styled from "styled-components"
import Artwork from "./artwork/grid_item"

interface Props extends RelayProps, React.HTMLProps<ArtworkGrid> {
  columnCount?: number
  sectionMargin?: number
  itemMargin?: number
  onLoadMore?: () => any
}

interface State {
  loading: boolean
  interval: any
}

export class ArtworkGrid extends React.Component<Props, State> {
  public static defaultProps: Partial<Props>

  constructor(props) {
    super(props)
    this.state = {
      interval: null,
      loading: false,
    }
  }

  componentDidMount() {
    if (this.props.onLoadMore) {
      const interval = setInterval(() => {
        this.maybeLoadMore()
      }, 150)
      this.setState({ interval })
    }
  }

  componentWillUnmount() {
    if (this.state.interval) {
      clearInterval(this.state.interval)
    }
  }

  maybeLoadMore() {
    const threshold = window.innerHeight + window.scrollY
    const el = ReactDOM.findDOMNode(this)
    if (threshold >= el.clientHeight + el.scrollTop) {
      this.props.onLoadMore()
    }
  }

  sectionedArtworks() {
    const sectionedArtworks: ArtworkRelayProps[][] = []
    const sectionRatioSums = []
    const artworks = this.props.artworks ? this.props.artworks.edges : []

    for (let i = 0; i < this.props.columnCount; i++) {
      sectionedArtworks.push([])
      sectionRatioSums.push(0)
    }

    artworks.forEach(artworkEdge => {
      const artwork = artworkEdge.node

      // There are artworks without images and other ‘issues’. Like Force we’re just going to reject those for now.
      // See: https://github.com/artsy/eigen/issues/1667
      if (artwork.image) {
        // Find section with lowest *inverted* aspect ratio sum, which is the shortest column.
        let lowestRatioSum = Number.MAX_VALUE
        let sectionIndex = null
        for (let j = 0; j < sectionRatioSums.length; j++) {
          const ratioSum = sectionRatioSums[j]
          if (ratioSum < lowestRatioSum) {
            sectionIndex = j
            lowestRatioSum = ratioSum
          }
        }

        if (sectionIndex != null) {
          const section = sectionedArtworks[sectionIndex]
          section.push(artwork)

          // Keep track of total section aspect ratio
          const aspectRatio = artwork.image.aspect_ratio || 1 // Ensure we never divide by null/0
          // Invert the aspect ratio so that a lower value means a shorter section.
          sectionRatioSums[sectionIndex] += 1 / aspectRatio
        }
      }
    })

    return sectionedArtworks
  }

  renderSections() {
    const spacerStyle = {
      height: this.props.itemMargin,
    }
    const sectionedArtworks = this.sectionedArtworks()
    const sections = []

    for (let i = 0; i < this.props.columnCount; i++) {
      const artworkComponents = []
      for (let j = 0; j < sectionedArtworks[i].length; j++) {
        const artwork = sectionedArtworks[i][j]
        artworkComponents.push(<Artwork artwork={artwork as any} key={"artwork-" + j + "-" + artwork.__id} />)
        // Setting a marginBottom on the artwork component didn’t work, so using a spacer view instead.
        if (j < sectionedArtworks[i].length - 1) {
          artworkComponents.push(<div style={spacerStyle} key={"spacer-" + j + "-" + artwork.__id} />)
        }
      }

      const sectionSpecificStyle = {
        flex: 1,
        minWidth: 0,
        marginRight: i === this.props.columnCount - 1 ? 0 : this.props.sectionMargin,
      }

      sections.push(<div style={sectionSpecificStyle} key={i}>{artworkComponents}</div>)
    }
    return sections
  }

  render() {
    const artworks = this.renderSections() || []
    return (
      <div className={this.props.className}>
        {artworks}
      </div>
    )
  }
}

ArtworkGrid.defaultProps = {
  columnCount: 3,
  sectionMargin: 20,
  itemMargin: 20,
}

const StyledGrid = styled(ArtworkGrid)`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
`

const ArtworkFragment = Relay.QL`
  fragment on Artwork {
    __id
    image {
      aspect_ratio
    }
    ${Artwork.getFragment("artwork")}
  }
`

export default Relay.createContainer(StyledGrid, {
  fragments: {
    artworks: () => Relay.QL`
      fragment on ArtworkConnection {
        edges {
          node {
            ${ArtworkFragment}
          }
        }
      }
    `,
  },
})

interface ArtworkRelayProps {
  __id: string
  image: {
    aspect_ratio: number | null
  } | null
}

interface RelayProps {
  artworks: {
    edges: Array<{
      node: any
    } | null> | null
  }
}
