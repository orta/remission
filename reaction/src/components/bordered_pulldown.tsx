import * as React from "react"

import Icon from "./icon"

import styled from "styled-components"
import colors from "../assets/colors"
import { secondary } from "../assets/fonts"

interface Props extends React.HTMLProps<BorderedPulldown> {
  options: any
  defaultValue: string
  onChange?: any
}

interface State {
  selected: any
  isHovered: boolean
}

export class BorderedPulldown extends React.Component<Props, State> {
  constructor(props) {
    super(props)
    this.state = {
      selected: null,
      isHovered: false,
    }
  }

  toggleHover(value) {
    this.setState({
      isHovered: value,
    })
  }

  onChange(option) {
    this.setState({
      selected: option,
      isHovered: false,
    })
    this.props.onChange(option)
  }

  render() {
    const { options, defaultValue } = this.props

    const optionEls = options.map(option => {
      return (
        <a key={option.val} onClick={() => this.onChange(option)}>
          {option.name}
        </a>
      )
    })

    const displayValue = (this.state.selected && this.state.selected.name) || defaultValue
    let pulldownStyles = {}

    if (this.state.isHovered) {
      pulldownStyles = {
        display: "block",
      }
    }

    return (
      <div
        className={this.props.className}
        onMouseEnter={() => this.toggleHover(true)}
        onMouseLeave={() => this.toggleHover(false)}
      >
        <Toggle>
          <span>
            {displayValue}
          </span>
          <CaretHolder>
            <Icon name="arrow-down" fontSize="9px" color={colors.grayMedium} />
          </CaretHolder>
        </Toggle>
        <PulldownOptions style={pulldownStyles}>
          {optionEls}
        </PulldownOptions>
      </div>
    )
  }
}

const StyledBorderedPulldown = styled(BorderedPulldown)`
  ${secondary.style}
  display: inline-block;
  width: 200px;
  position :relative;
  border: 2px solid ${colors.grayMedium};
  text-align: left;
  font-size: 17px;
  &.is-disabled {
    .bordered-pulldown-toggle {
      cursor: default;
      color: ${colors.grayBold};
    }
  }
`

const Toggle = styled.div`
  display: block;
  padding: 8px 10px 6px;
  text-decoration: none;
`

const CaretHolder = styled.div`
  float: right;
  padding-left: 5px;
  border-left: 1px solid ${colors.grayMedium};
`

const PulldownOptions = styled.div`
  display: none;
  position: absolute;
  background: white;
  border: 2px solid ${colors.grayMedium};
  top: -2px;
  left: -2px;
  right: -2px;
  z-index: 2;
  > a {
    text-decoration: none;
    overflow: ellipsis;
    display: block;
    padding: 8px 10px 6px;
    text-decoration: none;
    cursor: pointer;
    &:hover {
      background-color: ${colors.gray};
    } 
  } 
`

export default StyledBorderedPulldown
