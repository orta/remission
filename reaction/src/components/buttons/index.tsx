import Button from "./default"
import FacebookButton from "./facebook"
import GhostButton from "./ghost"
import InvertedButton from "./inverted"
import TwitterButton from "./twitter"

export default {
  Button,
  FacebookButton,
  GhostButton,
  InvertedButton,
  TwitterButton,
}
