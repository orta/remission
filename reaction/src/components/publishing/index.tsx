import Article from "./article"
import Header from "./header/header"
import RelatedArticlesCanvas from "./related_articles/related_articles_canvas"
import RelatedArticlesPanel from "./related_articles/related_articles_panel"
import Artwork from "./sections/artwork"
import Authors from "./sections/authors"
import Caption from "./sections/caption"
import Embed from "./sections/embed"
import FullscreenViewer from "./sections/fullscreen_viewer/fullscreen_viewer"
import Image from "./sections/image"
import ImageCollection from "./sections/image_collection"
import ImageSetPreview from "./sections/imageset_preview"
import ImageSetPreviewClassic from "./sections/imageset_preview_classic"
import Text from "./sections/text"
import Video from "./sections/video"

// Icon SVGs
import EditEmbed from "./icon/edit_embed"
import EditImages from "./icon/edit_images"
import EditText from "./icon/edit_text"
import EditVideo from "./icon/edit_video"
import HeroImage from "./icon/hero_image"
import HeroVideo from "./icon/hero_video"
import ImageFullscreen from "./icon/image_fullscreen"
import ImageSet from "./icon/image_set"
import LayoutFullscreen from "./icon/layout_fullscreen"
import LayoutSplit from "./icon/layout_split"
import LayoutText from "./icon/layout_text"
import Remove from "./icon/remove"

// Test Fixtures
import { ClassicArticle, FeatureArticle, StandardArticle } from "./fixtures/articles"

export default {
  Article,
  Artwork,
  Authors,
  Embed,
  Caption,
  FullscreenViewer,
  Header,
  Image,
  ImageCollection,
  ImageSetPreview,
  ImageSetPreviewClassic,
  RelatedArticlesCanvas,
  RelatedArticlesPanel,
  Text,
  Video,
  Icon: {
    EditEmbed,
    EditImages,
    EditText,
    EditVideo,
    HeroImage,
    HeroVideo,
    ImageFullscreen,
    ImageSet,
    LayoutFullscreen,
    LayoutSplit,
    LayoutText,
    Remove,
  },
  Fixtures: {
    ClassicArticle,
    FeatureArticle,
    StandardArticle,
  },
}
