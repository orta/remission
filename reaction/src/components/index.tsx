import Artwork from "./artwork"
import ArtworkFilter from "./artwork_filter"
import ArtworkGrid from "./artwork_grid"
import BorderedPulldown from "./bordered_pulldown"
import Buttons from "./buttons/index"
import Checkbox from "./checkbox"
import Gene from "./gene/index"
import Grid from "./grid"
import Icon from "./icon"
import Input from "./input"
import Modal from "./modal/modal"
import Nav from "./nav"
import Text from "./text"
import TextArea from "./text_area"
import TextLink from "./text_link"
import Title from "./title"

import * as React from "react"
import * as ReactDOM from "react-dom"
import * as Relay from "react-relay"

import { artsyNetworkLayer } from "../relay/config"
import * as Artsy from "./artsy"

interface InitOptions {
  user: User
  component: any
  domID: string
  queryConfig: any
}

export function init(options: InitOptions) {
  Relay.injectNetworkLayer(artsyNetworkLayer(options.user))

  const rootRoute = (
    <Artsy.ContextProvider currentUser={options.user}>
      <Relay.RootContainer Component={options.component} route={options.queryConfig} />
    </Artsy.ContextProvider>
  )

  ReactDOM.render(rootRoute, document.getElementById(options.domID))
}

export default {
  Artwork,
  ArtworkFilter,
  Buttons,
  Modal,
  ArtworkGrid,
  BorderedPulldown,
  Checkbox,
  Gene,
  Grid,
  Icon,
  Input,
  Nav,
  Text,
  TextArea,
  TextLink,
  Title,
}
