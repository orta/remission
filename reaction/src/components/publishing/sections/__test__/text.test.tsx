import * as React from "react"
import * as renderer from "react-test-renderer"
import { SectionText } from "../../fixtures/components"
import Text from "../text"

it("renders classic text properly", () => {
  const text = renderer.create(<Text html={SectionText.classic} layout="classic" />).toJSON()
  expect(text).toMatchSnapshot()
})

it("renders feature text properly", () => {
  const text = renderer.create(<Text html={SectionText.feature} layout="feature" />).toJSON()
  expect(text).toMatchSnapshot()
})

it("renders standard text properly", () => {
  const text = renderer.create(<Text html={SectionText.standard} layout="standard" />).toJSON()
  expect(text).toMatchSnapshot()
})
