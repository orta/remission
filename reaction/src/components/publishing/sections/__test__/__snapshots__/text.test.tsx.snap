// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders classic text properly 1`] = `
<div
  className="sc-bdVaJa vKJlg"
>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>Rines is a part of a recent migration that has galleries gravitating to the <a href='https://www.artsy.net/'>Lower East Side and Chinatown</a>. The new gallery sits on a stretch of Henry Street that also includes Chapter NY, Cuevas Tilleard, and newcomer Shrine, and is just blocks from:</p><ul><li>Chinatown stalwart Reena Spaulings Fine Art</li><li>The shiny new <span style='text-decoration: line-through;'>Foxy Production</span></li><li>Freshly launched programs like Erin Goldberger’s New Release</li><li>Angelo Lanza’s gallery Jeffrey Stark in the basement of the East Broadway Mall</li></ul><p>And <em>not a stone’s throw away</em> are the southern bounds of the Lower East Side, where <strong>hundreds</strong> of more galleries gather.</p><blockquote>Hardly a week goes by without murmurings that yet another gallery is opening</blockquote><h3><strong>Galleries Section, Booth 10221</strong></h3><h2>neugerriemschneider</h2><h3><em>With works by</em> Franz Ackermann, Ai Weiwei, Pawel Althamer, Billy Childish, Keith Edmier, Olafur Eliasson, Andreas Eriksson, Noa Eshkol, Mario García Torres, Renata Lucas, Michel Majerus, Mike Nelson, Jorge Pardo, Elizabeth Peyton, Tobias Rehberger, Thaddeus Strode, Rirkrit Tiravanija, Pae White</h3><p>The resultant work allows Salley the chance to recount her experiences of the aftermath of her scandal in her own words. In the film, Fujiwara and Salley are shown meeting professionals from public relations, advertising, and fashion companies as they seek to construct a new public image for her. Alongside the film, light boxes display fashion photographer Andreas Larsson’s pictures of Salley, which were taken as part of the project to rebuild her profile. While the show tackles public identity, female iconography, and Salley’s voice as an artist, the pair’s close working relationship—one in which the conventional power relationship has been overturned—no doubt aided their collaboration.</p><ol><li>Chinatown stalwart Reena Spaulings Fine Art</li><li>The shiny new Foxy Production</li><li>Freshly launched programs like Erin Goldberger’s New Release</li><li>Angelo Lanza’s gallery Jeffrey Stark in the basement of the East Broadway Mall</li></ol><p>Rines is a part of a recent migration that has galleries gravitating to the Lower East Side and Chinatown. She opened 56 Henry in December 2015, just four months after being forced to shutter her previous microgallery 55 Gansevoort—an elevator shaft a block east of the Whitney—after Restoration Hardware bought its building.</p><h2><a href='https://www.artsy.net/artist/ej-hill' class='is-follow-link' name='EJ_Hill'>EJ Hill</a><a data-id='ej-hill' class='entity-follow artist-follow'></a></h2><p><a href='https://www.artsy.net/artist/ej-hill' class='is-follow-link' name='EJ_Hill'>EJ Hill</a><a data-id='ej-hill' class='entity-follow artist-follow'></a></p>",
      }
    }
  />
</div>
`;

exports[`renders feature text properly 1`] = `
<div
  className="sc-bdVaJa gdoYdK"
>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<h1>Standard Header</h1><p><span class='content-start'>R</span>ines is a part of a recent migration that has galleries gravitating to the Lower East Side and Chinatown. She opened 56 Henry in December 2015, just four months after being forced to shutter her previous microgallery 55 Gansevoort—an elevator shaft a block east of the Whitney—after Restoration Hardware bought its building. The new gallery sits on a stretch of Henry Street that also includes Chapter NY, Cuevas Tilleard, and newcomer Shrine, and is just blocks from:</p><p>Rines is a part of a recent migration that has galleries gravitating to the <a href='https://www.artsy.net/'>Lower East Side and Chinatown</a>. The new gallery sits on a stretch of Henry Street that also includes Chapter NY, Cuevas Tilleard, and newcomer Shrine, and is just blocks from:</p><ul><li>Chinatown stalwart Reena Spaulings Fine Art</li><li>The shiny new <span style='text-decoration: line-through;'>Foxy Production</span></li><li>Freshly launched programs like Erin Goldberger’s New Release</li><li>Angelo Lanza’s gallery Jeffrey Stark in the basement of the East Broadway Mall</li></ul><p>And <em>not a stone’s throw away</em> are the southern bounds of the Lower East Side, where <strong>hundreds</strong> of more galleries gather.</p><blockquote>Hardly a week goes by without murmurings that yet another gallery is opening</blockquote><h3><strong>Galleries Section, Booth 10221</strong></h3><h2>neugerriemschneider</h2><h3><em>With works by</em> Franz Ackermann, Ai Weiwei, Pawel Althamer, Billy Childish, Keith Edmier, Olafur Eliasson, Andreas Eriksson, Noa Eshkol, Mario García Torres, Renata Lucas, Michel Majerus, Mike Nelson, Jorge Pardo, Elizabeth Peyton, Tobias Rehberger, Thaddeus Strode, Rirkrit Tiravanija, Pae White</h3><p>The resultant work allows Salley the chance to recount her experiences of the aftermath of her scandal in her own words. In the film, Fujiwara and Salley are shown meeting professionals from public relations, advertising, and fashion companies as they seek to construct a new public image for her. Alongside the film, light boxes display fashion photographer Andreas Larsson’s pictures of Salley, which were taken as part of the project to rebuild her profile. While the show tackles public identity, female iconography, and Salley’s voice as an artist, the pair’s close working relationship—one in which the conventional power relationship has been overturned—no doubt aided their collaboration.</p><ol><li>Chinatown stalwart Reena Spaulings Fine Art</li><li>The shiny new Foxy Production</li><li>Freshly launched programs like Erin Goldberger’s New Release</li><li>Angelo Lanza’s gallery Jeffrey Stark in the basement of the East Broadway Mall</li></ol><p>Rines is a part of a recent migration that has galleries gravitating to the Lower East Side and Chinatown. She opened 56 Henry in December 2015, just four months after being forced to shutter her previous microgallery 55 Gansevoort—an elevator shaft a block east of the Whitney—after Restoration Hardware bought its building.</p><h2><a href='https://www.artsy.net/artist/ej-hill' class='is-follow-link' name='EJ_Hill'>EJ Hill</a><a data-id='ej-hill' class='entity-follow artist-follow'></a></h2><p><a href='https://www.artsy.net/artist/ej-hill' class='is-follow-link' name='EJ_Hill'>EJ Hill</a><a data-id='ej-hill' class='entity-follow artist-follow'></a></p><p>Rines is a part of a recent migration that has galleries gravitating to the Lower East Side and Chinatown. She opened 56 Henry in December 2015, just four months after being forced to shutter her previous microgallery 55 Gansevoort—an elevator shaft a block east of the Whitney—after Restoration Hardware bought its building.<span class='content-end'></span></p>",
      }
    }
  />
</div>
`;

exports[`renders standard text properly 1`] = `
<div
  className="sc-bdVaJa jFIhsF"
>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>Rines is a part of a recent migration that has galleries gravitating to the <a href='https://www.artsy.net/'>Lower East Side and Chinatown</a>. The new gallery sits on a stretch of Henry Street that also includes Chapter NY, Cuevas Tilleard, and newcomer Shrine, and is just blocks from:</p><ul><li>Chinatown stalwart Reena Spaulings Fine Art</li><li>The shiny new <span style='text-decoration: line-through;'>Foxy Production</span></li><li>Freshly launched programs like Erin Goldberger’s New Release</li><li>Angelo Lanza’s gallery Jeffrey Stark in the basement of the East Broadway Mall</li></ul><p>And <em>not a stone’s throw away</em> are the southern bounds of the Lower East Side, where <strong>hundreds</strong> of more galleries gather.</p><blockquote>Hardly a week goes by without murmurings that yet another gallery is opening</blockquote><h3><strong>Galleries Section, Booth 10221</strong></h3><h2>neugerriemschneider</h2><h3><em>With works by</em> Franz Ackermann, Ai Weiwei, Pawel Althamer, Billy Childish, Keith Edmier, Olafur Eliasson, Andreas Eriksson, Noa Eshkol, Mario García Torres, Renata Lucas, Michel Majerus, Mike Nelson, Jorge Pardo, Elizabeth Peyton, Tobias Rehberger, Thaddeus Strode, Rirkrit Tiravanija, Pae White</h3><p>The resultant work allows Salley the chance to recount her experiences of the aftermath of her scandal in her own words. In the film, Fujiwara and Salley are shown meeting professionals from public relations, advertising, and fashion companies as they seek to construct a new public image for her. Alongside the film, light boxes display fashion photographer Andreas Larsson’s pictures of Salley, which were taken as part of the project to rebuild her profile. While the show tackles public identity, female iconography, and Salley’s voice as an artist, the pair’s close working relationship—one in which the conventional power relationship has been overturned—no doubt aided their collaboration.</p><ol><li>Chinatown stalwart Reena Spaulings Fine Art</li><li>The shiny new Foxy Production</li><li>Freshly launched programs like Erin Goldberger’s New Release</li><li>Angelo Lanza’s gallery Jeffrey Stark in the basement of the East Broadway Mall</li></ol><p>Rines is a part of a recent migration that has galleries gravitating to the Lower East Side and Chinatown. She opened 56 Henry in December 2015, just four months after being forced to shutter her previous microgallery 55 Gansevoort—an elevator shaft a block east of the Whitney—after Restoration Hardware bought its building.</p><h2><a href='https://www.artsy.net/artist/ej-hill' class='is-follow-link' name='EJ_Hill'>EJ Hill</a><a data-id='ej-hill' class='entity-follow artist-follow'></a></h2><p><a href='https://www.artsy.net/artist/ej-hill' class='is-follow-link' name='EJ_Hill'>EJ Hill</a><a data-id='ej-hill' class='entity-follow artist-follow'></a></p><p>Rines is a part of a recent migration that has galleries gravitating to the Lower East Side and Chinatown. She opened 56 Henry in December 2015, just four months after being forced to shutter her previous microgallery 55 Gansevoort—an elevator shaft a block east of the Whitney—after Restoration Hardware bought its building.<span class='content-end'></span></p>",
      }
    }
  />
</div>
`;
